edition = "2023";

package mmm.v1.model;

import "google/protobuf/any.proto";
import "mmm/v1/marketing/marketing_data.proto";

option java_multiple_files = true;

message MmmKernel {
  // The marketing data that is used to train the marketing mix model and
  // later analyzed by the model.
  marketing.MarketingData marketing_data = 1;

  // The details about the model implementation.
  //
  // This should contain a trained marketing mix model along with model-specific
  // information such as model convergence, flags of model usage.
  google.protobuf.Any model = 2;
}
